tests/unit/mixins/createI18nMixin.ts

```diff
@@ -0,0 +1,353 @@
+import * as registerSuite from 'intern!object';
+import * as assert from 'intern/chai!assert';
+import createStateful from 'dojo-compose/mixins/createStateful';
+import i18n, { Bundle, invalidate, Messages, switchLocale, systemLocale } from 'dojo-i18n/i18n';
+import createI18nMixin from '../../../src/mixins/createI18nMixin';
+import bundle from '../../support/nls/greetings';
+
+const createLocalized = createStateful
+	.mixin({
+		mixin: createI18nMixin,
+		aspectAdvice: {
+			before: {
+				registerBundle(this: any, bundle: Bundle<Messages>) {
+					if (!this.registeredBundles) {
+						this.registeredBundles = [];
+					}
+					this.registeredBundles.push(bundle);
+					return [ bundle ];
+				}
+			}
+		}
+	});
+let localized: any;
+
+registerSuite({
+	name: 'mixins/createI18nMixin',
+
+	afterEach() {
+		switchLocale(systemLocale);
+		invalidate();
+
+		if (localized) {
+			localized.destroy();
+			localized = null;
+		}
+	},
+
+	api() {
+		const localized = createLocalized();
+		assert(localized);
+		assert.isFunction(localized.localizeBundle);
+		assert.isFunction(localized.registerBundle);
+	},
+
+	'.localizeBundle()/.registerBundle()': {
+		'Returns default messages when locale bundle not loaded'() {
+			switchLocale('fr');
+			localized = createLocalized();
+			const messages = localized.localizeBundle(bundle);
+
+			assert.strictEqual(messages['hello'], 'Hello');
+			assert.strictEqual(messages['goodbye'], 'Goodbye');
+		},
+
+		'Uses `state.locale` when available'() {
+			localized = createLocalized({
+				state: {
+					locale: 'fr'
+				}
+			});
+			return i18n(bundle, localized).then(function () {
+				const messages = localized.localizeBundle(bundle);
+				assert.strictEqual(messages['hello'], 'Bonjour');
+				assert.strictEqual(messages['goodbye'], 'Au revoir');
+			});
+		},
+
+		'Uses the parent\'s `state.locale` when available'() {
+			localized = createLocalized();
+			(<any> localized).parent = <any> {
+				state: { locale: 'fr' }
+			};
+			return i18n(bundle, 'fr').then(function () {
+				const messages = localized.localizeBundle(bundle);
+				assert.strictEqual(messages['hello'], 'Bonjour');
```

Comment:
It can be; I didn't realize TS had been updated to allow dot notation on `[key: string]` interfaces (previously the threw warnings with dot notation).